tests/app/questionnaire/test_router.py

```diff
@@ -216,15 +218,40 @@ def test_return_to_final_summary_next_location_url(self):
 
         self.assertEqual(next_location, expected_location)
 
+    def test_last_block_on_path_routes_to_first_incomplete_location_in_section_when_section_not_complete(
+        self
+    ):
+        schema = Mock()
+        schema.get_block.return_value = {"type": "Question"}
+        progress_store = ProgressStore(
+            [
+                {
+                    "section_id": "section-1",
+                    "list_item_id": None,
+                    "status": CompletionStatus.IN_PROGRESS,
+                    "block_ids": ["block-1"],
+                }
+            ]
+        )
+        router = Router(
+            schema, self.answer_store, self.list_store, progress_store, self.metadata
+        )
+        current_location = Location(section_id="section-1", block_id="block-1")
+        routing_path = RoutingPath(
+            ["block-1", "block-2", "block-1"], section_id="section-1"
```

Comment:
Is this a valid path?

Reply:
Yes, this recreates our ccs schemas behaviour where you are forced to go back to primary person list collector if there is no primary person the first time round.